modules/objdetect/src/qrcode.cpp

```diff
@@ -1973,6 +1973,29 @@ vector<vector<float> > QRDecode::computeSpline(const vector<int> &x_arr, const v
     return S;
 }
 
+// get horizontal_order and update x_arr, y_arr if necessary
+static inline bool setHorizontalOrder(vector<int>& x_arr, vector<int>& y_arr) {
+    bool horizontal_order = abs(x_arr.front() - x_arr.back()) > abs(y_arr.front() - y_arr.back());
+    int countX = 0, countY = 0;
+    for (size_t i = 1ull; i < x_arr.size(); i++) {
+        if (x_arr[i] == x_arr[i-1])
+            countX++;
+        if (y_arr[i] == y_arr[i-1])
+            countY++;
+    }
+    if (countX != 0 || countY != 0) {
+        vector<int>& newX = countX < countY ? x_arr : y_arr;
+        const int delta = newX.back() > newX.front() ? 1 : -1;
+        horizontal_order = countX < countY ? true : false;
+        if (min(countX, countY) > 0)
+            for (size_t i = 1ull; i < x_arr.size(); i++) {
+                if (newX[i] == newX[i-1])
+                    newX[i] += delta;
+            }
```

Comment:
This hack requires description right in the code. Looks like some array values are modified on +-1, but the reason and idea is not obvious.

Reply:
Added comment

```
Only the points of the function can be interpolated using splines.
Contour points may not be a function, there may be several points with the same x coordinate and different y.
We are adding a small offset in x coord to fix this problem. This should not significantly affect the final answer.
```